### PR TITLE
add FailedAsyncInferWithNegativeTimeForWait test case

### PR DESCRIPTION
**Pull request by panhaiqi**
### Details:
 - *FailedAsyncInferWithNegativeTimeForWait*
 
### Tickets:
 - *CVS-126490*

### CODE REVIEW

**Comment by peterchen-intel** [comment]
$ grep -e "ov_cpu_func_tests-0 INFO:\s.*\sOK.*" -o test/ov_cpu_func_tests-0.log | grep OVInferRequestWaitTests.* -o
OVInferRequestWaitTests.CorrectOneAsyncInferWithGetInOutWithInfWait/targetDevice=CPU_ (4 ms)
OVInferRequestWaitTests.CorrectOneAsyncInferWithGetInOutWithInfWait/targetDevice=CPU_configItem=CPU_THROUGHPUT_STREAMS_CPU_THROUGHPUT_AUTO (6 ms)
OVInferRequestWaitTests.CorrectOneAsyncInferWithGetInOutWithInfWait/targetDevice=CPU_configItem=CPU_THREADS_NUM_1configItem=CPU_THROUGHPUT_STREAMS_0 (5 ms)
OVInferRequestWaitTests.canstart_asyncInferWithGetInOutWithStatusOnlyWait/targetDevice=CPU_ (7 ms)
OVInferRequestWaitTests.canstart_asyncInferWithGetInOutWithStatusOnlyWait/targetDevice=CPU_configItem=CPU_THROUGHPUT_STREAMS_CPU_THROUGHPUT_AUTO (14 ms)
OVInferRequestWaitTests.canstart_asyncInferWithGetInOutWithStatusOnlyWait/targetDevice=CPU_configItem=CPU_THREADS_NUM_1configItem=CPU_THROUGHPUT_STREAMS_0 (10 ms)
OVInferRequestWaitTests.canWaitWithotStartSsync/targetDevice=CPU_ (5 ms)
OVInferRequestWaitTests.canWaitWithotStartSsync/targetDevice=CPU_configItem=CPU_THROUGHPUT_STREAMS_CPU_THROUGHPUT_AUTO (5 ms)
OVInferRequestWaitTests.canWaitWithotStartSsync/targetDevice=CPU_configItem=CPU_THREADS_NUM_1configItem=CPU_THROUGHPUT_STREAMS_0 (4 ms)
OVInferRequestWaitTests.throwExceptionOnSetTensorAfterAsyncInfer/targetDevice=CPU_ (4 ms)
OVInferRequestWaitTests.throwExceptionOnSetTensorAfterAsyncInfer/targetDevice=CPU_configItem=CPU_THROUGHPUT_STREAMS_CPU_THROUGHPUT_AUTO (5 ms)
OVInferRequestWaitTests.throwExceptionOnGetTensorAfterAsyncInfer/targetDevice=CPU_ (4 ms)
OVInferRequestWaitTests.throwExceptionOnGetTensorAfterAsyncInfer/targetDevice=CPU_configItem=CPU_THROUGHPUT_STREAMS_CPU_THROUGHPUT_AUTO (5 ms)
OVInferRequestWaitTests.throwExceptionOnGetTensorAfterAsyncInfer/targetDevice=CPU_configItem=CPU_THREADS_NUM_1configItem=CPU_THROUGHPUT_STREAMS_0 (4 ms)
OVInferRequestWaitTests.FailedAsyncInferWithNegativeTimeForWait/targetDevice=CPU_ (5 ms)
OVInferRequestWaitTests.FailedAsyncInferWithNegativeTimeForWait/targetDevice=CPU_configItem=CPU_THROUGHPUT_STREAMS_CPU_THROUGHPUT_AUTO (6 ms)
OVInferRequestWaitTests.FailedAsyncInferWithNegativeTimeForWait/targetDevice=CPU_configItem=CPU_THREADS_NUM_1configItem=CPU_THROUGHPUT_STREAMS_0 (5 ms)

File: src/plugins/intel_gna/tests/functional/shared_tests_instances/skip_tests_config.cpp [hunk 1]
```diff
@@ -107,5 +104,10 @@ std::vector<std::string> disabledTestPatterns() {
         R"(.*smoke_Decompose2DConv.*)",
         // TODO: Issue: 123306
         R"(smoke_convert_matmul_to_fc/ConvertMatmulToFcWithTransposesPass.CompareWithRefImpl/netPRC=FP(32|16)_targetDevice=GNA__configItem=GNA_COMPACT_MODE_NO_configItem=GNA_DEVICE_MODE_GNA_SW_(FP32|EXACT)_IS=\(8.*)",
+        // TODO: Issue: 125686
```
Comment (peterchen-intel) [inline src/plugins/intel_gna/tests/functional/shared_tests_instances/skip_tests_config.cpp:107]:
```suggestion
        // TODO: CVS-125686
```

File: src/tests/functional/plugin/shared/src/behavior/ov_infer_request/wait.cpp [hunk 2]
```diff
@@ -17,6 +17,10 @@ void OVInferRequestWaitTests::SetUp() {
     output = execNet.output();
 }
 
+std::string OVInferRequestWaitTests::getTestCaseName(testing::TestParamInfo<InferRequestParams> obj) {
+    return OVInferRequestTests::getTestCaseName(obj);
+}
+
```
[the remark this answers is not in the document]
Comment (riverlijunjie) [reply src/tests/functional/plugin/shared/src/behavior/ov_infer_request/wait.cpp:23]: It is strange that if we don't add this `getTestCaseName`, the functions' symbols of `OVInferRequestWaitTests` will not to added into `ov_cpu_func_tests` or other plugin's func test application, which cause `OVInferRequestWaitTests` cannot run. And we also dumped the symbols of `libfuncSharedTests.a`, all symbols of `OVInferRequestWaitTests` can be found.
Could you give some hints on why it was?

File: src/tests/functional/plugin/shared/include/behavior/ov_infer_request/wait.hpp [hunk 3]
```diff
@@ -11,6 +11,7 @@ namespace test {
 namespace behavior {
 struct OVInferRequestWaitTests : public OVInferRequestTests {
     void SetUp() override;
+    static std::string getTestCaseName(testing::TestParamInfo<InferRequestParams> obj);
```
Comment (peterchen-intel) [inline src/tests/functional/plugin/shared/include/behavior/ov_infer_request/wait.hpp:14]: @panhaiqi  Please create a ticket to follow and put the TODo: number here.

**Review by vurusovs — APPROVED** [review]

**Pull request merged**